inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/tube/InLongTubeFetcherImpl.java

```diff
@@ -0,0 +1,291 @@
+                                inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
+                        .addCallbackTimeCost(System.currentTimeMillis() - start).addCallbackDoneTimes(1);
+            } catch (Exception e) {
+                context.getStatManager()
+                        .getStatistics(context.getConfig().getSortTaskId(),
+                                inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
+                        .addCallbackErrorTimes(1);
+                e.printStackTrace();
+            }
+        }
+
+        /**
+         * parseAttr from k1=v1&k2=v2 to kv map
+         *
+         * @param splitter {@link Splitter}
+         * @param attr String
+         * @param entrySplitterStr String
+         * @return {@link Map<String,String>}
+         */
+        private Map<String, String> parseAttr(Splitter splitter, String attr, String entrySplitterStr) {
+            Map<String, String> map = new HashMap<>();
+            for (String s : splitter.split(attr)) {
+                int idx = s.indexOf(entrySplitterStr);
+                String k = s;
+                String v = null;
+                if (idx > 0) {
+                    k = s.substring(0, idx);
+                    v = s.substring(idx + 1);
+                }
+                map.put(k, v);
+            }
+            return map;
+        }
+
+        private Map<String, String> getAttributeMap(String attribute) {
+            final Splitter splitter = Splitter.on("&");
+            return parseAttr(splitter, attribute, "=");
+        }
+
+        @Override
+        public void run() {
+            boolean hasPermit;
+            while (true) {
+                hasPermit = false;
+                try {
+                    if (context.getConfig().isStopConsume() || stopConsume) {
+                        TimeUnit.MILLISECONDS.sleep(50);
+                        continue;
+                    }
+
+                    if (sleepTime > 0) {
+                        TimeUnit.MILLISECONDS.sleep(sleepTime);
+                    }
+
+                    context.acquireRequestPermit();
+                    hasPermit = true;
+                    context.getStatManager()
+                            .getStatistics(context.getConfig().getSortTaskId(),
+                                    inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
+                            .addMsgCount(1).addFetchTimes(1);
+
+                    long startFetchTime = System.currentTimeMillis();
+                    ConsumerResult message = messageConsumer.getMessage();
+                    context.getStatManager()
+                            .getStatistics(context.getConfig().getSortTaskId(),
+                                    inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
+                            .addFetchTimeCost(System.currentTimeMillis() - startFetchTime);
```

Comment:
In a loop turn, it is repeat invoking for "System.currentTimeMillis()".
long startFetchTime = System.currentTimeMillis();

Reply:
Need to count method call time